feat: [P4ADEV-2074] multi filter

mmoio opened the pull request:
<!--- Please always add a PR description as if nobody knows anything about the context these changes come from. -->
<!--- Even if we are all from our internal team, we may not be on the same page. -->
<!--- Write this PR as you were contributing to a public OSS project, where nobody knows you and you have to earn their trust. -->
<!--- This will improve our projects in the long run! Thanks. -->

#### List of Changes

<!--- Describe your changes in detail -->
- Adds multi-filter functionality
- Creates new FormComponent and its sub-components
- Renames DateRange component to FormComponent/_DateRange
- Creates new MultiFilter component and its sub-components
- Modifies FilterContainer to use new FormComponent and its sub-components
- Modifies SearchCard and Treasury components to use new MultiFilter component
- Adds new useFilters hook
- Adds new FilterStore store
- Modifies GlobalStore and its test to use new FilterStore
- Updates translations files
- Unit tests
#### Motivation and Context

<!--- Why is this change required? What problem does it solve? -->
Adds multifilter functionality and updates files that were using the previous version
#### How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, tests ran to see how -->
<!--- your change affects other areas of the code, etc. -->
- unit tests
- visual testing
#### Screenshots (if appropriate):
![image](https://github.com/user-attachments/assets/bb6d9a6a-d063-4c90-b626-a4e2e5c8372d)

#### Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

mmoio committed:
feat: [P4ADEV-2074] multi filter

- Adds multi-filter functionality
- Creates new FormComponent and its sub-components
- Renames DateRange component to FormComponent/_DateRange
- Creates new MultiFilter component and its sub-components
- Modifies FilterContainer to use new FormComponent and its sub-components
- Modifies SearchCard and Treasury components to use new MultiFilter component
- Adds new useFilters hook
- Adds new FilterStore store
- Modifies GlobalStore and its test to use new FilterStore
- Updates translations and yarn.lock files`
- Unit tests

vito80ba commented on src/components/FormComponent/_Select.tsx:
```diff
@@ -0,0 +1,31 @@
+  const onChange = (e: ChangeEvent<HTMLInputElement>) => setValue(e.target.value);
+
+  return (
+    <TextField
```
Why a TextField to render a `Select box` ?

mmoio replied:
No reason really, just personal preference because is less code and the label is handled by one component. It still uses Mui Select component under the hood, but i can change it to a Select

vito80ba commented on src/components/FormComponent/_Select.tsx:
```diff
@@ -0,0 +1,31 @@
+      value={value}
+      onChange={onChange}
+      {...props}
+      select>
```
Take a look to the console, it seems raise error
 > Received `true` for a non-boolean attribute `select`

mmoio replied:
Seems fixed with the latest commit

mmoio committed:
feat: [P4ADEV-2074] multi filter

- filter type cannot be used more then once
- refactors filters state
- Select component accept all mui OptionProps
- add unit test for the FilterStore

mmoio committed:
fix: [P4ADEV-2074] filters fixes

- clean filters onMount
- disable SearchCard filter
  button on no filter selected
- fixes GlobalStore test file
